target/linux/realtek/image/rtl839x.mk

```diff
@@ -11,6 +11,17 @@ define Device/panasonic_m48eg-pn28480k
 endef
 TARGET_DEVICES += panasonic_m48eg-pn28480k
 
+define Device/tplink_sg2452p-v4
+  SOC := rtl8393
+  KERNEL_SIZE := 6m
+  IMAGE_SIZE := 26m
+  DEVICE_VENDOR := TP-Link
+  DEVICE_MODEL := SG2452P
+  DEVICE_VARIANT := v4
```

Comment:
You could add `DEVICE_ALT0_*` variables to indicate the T1600G-52PS v4 is really the same device. Do you where  TP-Link uses these names? Or are both used as model names on the device label?

Reply:
On the front panel it reads T1600G-52PS, on the bottom it reads both names. I opted for Tl-SG2452 to be consistent with the naming of the other TP-Link devices supported on the Realtek target.